docs/Hosting.md

```diff
@@ -0,0 +1,417 @@
+* Attackers could scrape static data for amplification. → Mitigation: minimal static set, rate limiting.
+* Accidental production enablement. → Mitigation: require BOTH flag and env var (defense in depth) OR build-time compile symbol for distribution images.
+
+### 6.8 Ideal Entra ID (Authorization Code + PKCE + OBO) Flow & Gaps
+Target future architecture for per-user delegated access:
+
+**Target Sequence**
+1. MCP server advertises (in its Protocol Registration / PRM document) that it uses Microsoft Entra ID for auth (includes issuer / audience metadata pointers or discovery hints).
+2. Client connects to MCP server and retrieves the PRM document.
+3. Client discovers Entra as the Authorization Server (AS) from PRM metadata.
+4. Client constructs Entra OpenID Connect / OAuth2 metadata URLs (well-known discovery) and fetches endpoints.
+5. Client registers with the Authorization Server (dynamic client registration) OR already possesses a suitable client ID.
+6. Client initiates Authorization Code Flow with PKCE directly against Entra (user authenticates + consents).
+7. Entra issues an access token whose `aud` (audience) is the MCP server (plus optional ID token for client state).
+8. Client sends that access token to the MCP server on each request (e.g., Bearer token header / agreed channel field).
+9. MCP server validates token (issuer, signature, expiry, audience, scopes / roles, nonce / c_hash as needed).
+10. MCP server performs OAuth 2.0 On-Behalf-Of (OBO) flow to exchange the user token for downstream resource tokens (ARM, Storage, etc.).
+11. MCP server maps/associates downstream tokens to the inbound user principal (cache keyed by user object ID + scopes) and executes tool operations.
+12. Responses returned; auditing & telemetry capture per-user context.
+
+**Current Blocking Gap**: No dynamic client registration for arbitrary third-party native MCP clients.
+
+**Interim Alternatives**
+| Approach | Description | Pros | Cons / Risks |
+|----------|-------------|------|--------------|
+| Pre-registered multi-tenant Public Client | Publish a client ID (no secret) that all MCP clients can use for auth code + PKCE | Simple for ecosystem, no secret distribution | Hard to apply client-specific policies, revocation affects all, potential misuse rate limits |
+| Per-vendor Client IDs (manual registration) | Each MCP client vendor maintains an Entra app registration | Stronger isolation/accountability | Friction for new clients; coordination overhead |
+| Device Code Flow (interactive fallback) | Clients invoke device code to let user authenticate in browser | Works without embedded browser, simpler UX for CLIs | Less seamless, slower, not ideal for high-frequency re-auth scenarios |
+| Managed Identity Only (current preview baseline) | No per-user token; server uses MI for downstream calls | Simplifies backend, no OBO complexity | Loses per-user RBAC & auditing granularity; potential over-privileged MI |
```

Comment:
But how will that work in practice? For example, I connect your MCP server to my client. How does it know what identity to use for my use-case?